scapy/contrib/ospf.py

```diff
@@ -126,47 +127,26 @@ def guess_payload_class(self, payload):
 
 class LLS_Generic_TLV(Packet):
     name = "LLS Generic"
-    fields_desc = [ShortField("type", 1),
-                   FieldLenField("len", None, length_of=lambda x: x.val),
+    fields_desc = [ShortField("type", 0),
+                   FieldLenField("len", None, length_of="val"),
                    StrLenField("val", "", length_from=lambda x: x.len)]
 
     def guess_payload_class(self, p):
         return conf.padding_layer
 
-
-class LLS_ExtendedOptionsField(FlagsField):
-
-    def __init__(self, name="options", default=0, size=32,
-                 names=None):
-        if names is None:
-            names = ["LR", "RS"]
-        FlagsField.__init__(self, name, default, size, names)
-
-
 class LLS_Extended_Options(LLS_Generic_TLV):
     name = "LLS Extended Options and Flags"
     fields_desc = [ShortField("type", 1),
-                   ShortField("len", 4),
-                   LLS_ExtendedOptionsField()]
-
+                   FieldLenField("len", None, fmt="!H", length_of="options"),
+                   StrLenField("options", "", length_from=lambda x: x.len)]
+                 # TODO: FlagsField("options", 0, names=["LR", "RS"], size) with dynamic size
```

Comment:
Is this needed?

Reply:
We don't have a way (or at least i don't know about it) to have dynamic sized FlagsField (it would look such as `FlagsField("options", 0, names..., length_from=lambda pkt: pkt.len)`)

Current implementation was wrong because this size may vary, which was not taken into account